resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -95,6 +105,23 @@ void query() {
         sendQuery(query);
     }
 
+    static void addRecord(DatagramDnsQuery query, DnsSection section, DnsRecord record) {
+        if (record == null) {
+            return;
+        }
+        query.addRecord(section, record);
+    }
+
+    static void addRecords(DatagramDnsQuery query, DnsSection section, List<DnsRecord> records) {
+        if (records == null) {
+            return;
+        }
+
+        for (DnsRecord r : records) {
```

Comment:
The Netty usual coding style tends to avoid creating Iterators on the hot path, instead of relying on the JIT's escape analysis to kick off after some time. So I'm pretty sure you'll be asked to switch to a `for (int i = 0; records.size(); i++)` loop.